src/ReachSets/ContinuousPost/BFFPS19/reach.jl

```diff
@@ -0,0 +1,272 @@
+    end
+    push!(args, overapproximate_fun)
+
+    # overapproximate function for inputs
+    lazy_inputs_interval = options[:lazy_inputs_interval]
+    if lazy_inputs_interval == lazy_inputs_interval_always
+        overapproximate_inputs_fun = (k, i, x) -> overapproximate_fun(i, x)
+    else
+        # first set in a series
+        function _f(k, i, x::LinearMap{NUM}) where {NUM}
+            @assert k == 1 "a LinearMap is only expected in the first iteration"
+            return CacheMinkowskiSum(LazySet{NUM}[x])
+        end
+        # further sets of the series
+        function _f(k, i, x::MinkowskiSum{NUM, <:CacheMinkowskiSum}) where NUM
+            if has_constant_directions(block_options_iter, i)
+                # forget sets if we do not use epsilon-close approximation
+                forget_sets!(x.X)
+            end
+            push!(array(x.X), x.Y)
+            if lazy_inputs_interval(k)
+                # overapproximate lazy set
+                y = overapproximate_fun(i, x.X)
+                return CacheMinkowskiSum(LazySet{NUM}[y])
+            end
+            return x.X
+        end
+        function _f(k, i, x)
+            # other set types
+            if lazy_inputs_interval(k)
+                # overapproximate lazy set
+                return overapproximate_fun(i, x.X)
+            end
+            return x
+        end
+        overapproximate_inputs_fun = _f
+    end
+    push!(args, overapproximate_inputs_fun)
+
+    # ambient dimension
+    push!(args, n)
+
+    # number of computed sets
+    push!(args, N)
+
+    # output function
+    push!(args, output_function)
+
+    # add mode-specific block(s) argument
+    push!(args, blocks)
+    push!(args, partition)
+
+    # time step
+    push!(args, options[:δ])
+
+    # termination function
+    invariant = stateset(problem.s)
+    vars = get_vars(blocks, partition)
+    if (invariant == Universe(dim(invariant)))
+        invariant = Universe(length(vars))
+    else
+        invariant = LazySets.Approximations.project(invariant, vars)
+    end
+    termination = get_termination_function_out(N, invariant)
+    block_options = options[:block_options_init]
+    push!(args, constraints_list)
+    push!(args, block_options)
+    push!(args, vars)
+    push!(args, termination)
+
+    info("- Computing successors")
+
+    # progress meter
+    progress_meter = (N != nothing) ?
+        Progress(N, 1, "Computing successors ") :
+        nothing
+    push!(args, progress_meter)
+
+    # choose algorithm backend
+    algorithm = options[:algorithm]
+    if algorithm == "explicit"
+        algorithm_backend = "explicit_blocks_mixed"
+    elseif algorithm == "wrap"
+        algorithm_backend = "wrap"
+    else
+        error("Unsupported algorithm: ", algorithm)
+    end
+    push!(args, res)
+
+    # call the adequate function with the given arguments list
+    @timing begin
+        index, skip = available_algorithms[algorithm_backend]["func"](args...)
```

Comment:
(See the previous comment.)
```suggestion
        index, skip = reach_blocks!(args...)
```